src/WebJobs.Script/OutOfProc/Rpc/RpcInitializationService.cs

```diff
@@ -78,7 +79,9 @@ public async Task StartAsync(CancellationToken cancellationToken)
         public async Task StopAsync(CancellationToken cancellationToken)
         {
             _logger.LogDebug("Shuttingdown Rpc Channels Manager");
+            System.Timers.Timer disposeTimer = GetAndStartBufferDrainTimer();
```

Comment:
Can you please add comment here why this is needed? 
Adding comment to follow up offline. Since shutting down RpcServer will wait for disposing Jobhost which disposes function dipatcher, may be we do not needed this?

Reply:
Yes, but the part of code that waits for those sequence of dispose events is the OuterStopAsync() (also where the rpc server shutdown happens). StopAsync() in RpcInitializationService() is still invoked first. So, either timer needs to be added here or _webhostLanguageChannelManager.ShutdownChannels needs to move to OuterStopAsync().

We can chat more offline if I'm missing something else here.